columnq/src/table/mod.rs

```diff
@@ -149,23 +227,59 @@ impl Default for TableOptionParquet {
 pub struct TableOptionDelta {
     #[serde(default = "TableOptionDelta::default_use_memory_table")]
     use_memory_table: bool,
+    #[serde(default = "TableOptionDelta::default_batch_size")]
+    batch_size: usize,
 }
 
 impl TableOptionDelta {
     #[inline]
     pub fn default_use_memory_table() -> bool {
         true
     }
+
+    #[inline]
+    pub fn default_batch_size() -> usize {
+        1024
+    }
+
+    #[inline]
+    pub fn with_use_memory_table(mut self, use_memory_table: bool) -> Self {
+        self.use_memory_table = use_memory_table;
+        self
+    }
 }
 
 impl Default for TableOptionDelta {
     fn default() -> Self {
         Self {
             use_memory_table: Self::default_use_memory_table(),
+            batch_size: Self::default_batch_size(),
         }
     }
 }
 
+
+#[derive(Deserialize, Debug, Clone, Eq, PartialEq)]
+pub struct TableOptionNdjson {
+    #[serde(default = "TableOptionNdjson::default_batch_size")]
+    batch_size: usize,
```

Comment:
since `batch_size` is not a format specific option, but a datafusion query execution knob that applies to all formats, how about we move it into a filed in `TableSource`? This way, we won't need to reimplement this field for every format specific load option.